### PR TITLE
CC-331: update config options docs

### PR DESCRIPTION
**Pull request by shikhar**
Rewrote some of the config option docs for clarity and addressed comments from https://github.com/confluentinc/kafka-connect-elasticsearch/pull/31

### CODE REVIEW

**Comment by shikhar** [comment]
@confluentinc/connect for review

File: docs/configuration_options.rst [hunk 1]
```diff
@@ -1,100 +1,110 @@
...
 
   * Type: long
-  * Default: 10000
+  * Default: 1
   * Importance: low
 
-``linger.ms``
-  The task groups together any records that arrive in between request transmissions into a single batched request. Normally this occurs only under load when records arrive faster than they can be sent out. However in some circumstances the tasks may want to reduce the number of requests even under moderate load. This setting accomplishes this by adding a small amount of artificial delay. Rather than immediately sending out a record the task will wait for up to the given delay to allow other records to be sent so that the sends can be batched together.
+``flush.timeout.ms``
+  The timeout in milliseconds when flushing records to Elasticsearch.
```
Comment (shikhar) [inline docs/configuration_options.rst]: Seems worth documenting to me, will add a note (the task will fail, no retries).

File: docs/configuration_options.rst [hunk 2]
```diff
@@ -1,100 +1,110 @@
...
 
   * Type: int
-  * Default: 20000
-  * Importance: low
+  * Default: 5
+  * Importance: medium
 
 ``max.retries``
-  The max allowed number of retries.
+  The maximum number of retries that are allowed.
```
Comment (hachikuji) [inline docs/configuration_options.rst]: Similar to the previous comment. What happens when the retry limit is reached?

Comment (shikhar) [inline docs/configuration_options.rst]: Task failure, will doc

File: docs/configuration_options.rst [hunk 3]
```diff
@@ -1,100 +1,110 @@
...
 
   * Type: int
   * Default: 5
   * Importance: low
 
 ``retry.backoff.ms``
-  The amount of time to wait before attempting to retry a failed batch. This avoids repeatedly sending requests in a tight loop under some failure scenarios.
+  How long to wait in milliseconds, before attempting to retry a failed batch. This avoids retrying requests in a tight loop under certain failure scenarios.
```
Comment (hachikuji) [inline docs/configuration_options.rst]: The way this is phrased makes it sound like there are some failure scenarios where we _will_ retry requests in a tight loop. Is that intentional?

**Comment by hachikuji** [comment]
Minor comments, but looks good overall.

File: docs/configuration_options.rst (hunk 3 above)
Comment (shikhar) [inline docs/configuration_options.rst]: Unintentional, will rephrase

Comment (hachikuji) [inline docs/configuration_options.rst]: One other nit: the comma in the first sentence seems unneeded.

**Comment by hachikuji** [comment]
LGTM

**Pull request merged**